### ISSUE TITLE
Client throws NoSuchMethodError on unexpected primary data

### ISSUE THREAD

**Issue by otopba** [f3ath/json-api-dart#102]
Hi!
Method `deleteResourceAt` passes null `decodePrimaryData` to `_call`.
So if server will return something on DELETE request we will have exception inside  `Document.fromJson` function

![image](https://user-images.githubusercontent.com/4225279/83928645-1efdb600-a799-11ea-940c-5dce20bff268.png)
![image](https://user-images.githubusercontent.com/4225279/83928696-481e4680-a799-11ea-89a7-8232d0ad6364.png)

**Comment by f3ath** [comment]
The standard does not allow returning primary data on [resource deletion](https://jsonapi.org/format/#crud-deleting-responses) except for the 202 Accepted response. Other than that, there may be 3 general cases:
- no content
- error response
- meta data

neither of which is expected to have primary data. So I'm not sure if I understand the issue here. Would you mind providing an example of server response?

**Comment by otopba** [comment]
@f3ath You are right about the standard. But maybe the library should handle this response correctly. Because now it throws NoSuchMethodError

Of course, this is the wrong server behavior. But it seems that the client should not break in this case

Example of the response:

```json
{
   "data":{
      "id":"4f07091d-cbde-4475-8639-031fafa785ea",
      "type":"post"
   }
}
```

**Comment by f3ath** [comment]
Yep that's a perfectly valid point.

**Comment by f3ath** [comment]
A null check should be sufficient then. @otopba do you have an opinion on how this case should be handled?

**Comment by otopba** [comment]
@f3ath 
I think that the library can ignore the body of the response in this case. Because this answer is not made by standard

**Comment by f3ath** [comment]
Fix released in v4.2.2. @otopba please confirm the issue is resolved.

**Comment by otopba** [comment]
@f3ath Thank you. Now it works great 🤗
